opentelemetry/proto/profiles/v1development/profiles.proto

```diff
@@ -463,13 +463,17 @@ message Line {
 // Describes a function, including its human-readable name, system name,
 // source file, and starting line number in the source.
 message Function {
-  // Name of the function, in human-readable form if available.
-  int32 name_strindex = 1; // Index into string table
-  // Name of the function, as identified by the system.
-  // For instance, it can be a C++ mangled name.
-  int32 system_name_strindex = 2; // Index into string table
-  // Source file containing the function.
-  int32 filename_strindex = 3; // Index into string table
+  // Index into Profile.string_table to the name of the function, in
+  // human-readable form.
+  // Use index 0 if value is not available.
+  int32 name_strindex = 1;
+  // Index into Profile.string_table to the name of the function, as identified
+  // by the system. For instance, it can be a C++ mangled name.
+  // Use index 0 if value is not available.
+  int32 system_name_strindex = 2;
+  // Index into Profile.string_table to the source file containing the function.
+  // Use index 0 if value is not available.
+  int32 filename_strindex = 3;
   // Line number in source file.
   int64 start_line = 4;
```

Comment:
Can we make a note here? 0 if not available/making it optional?

Reply:
My understanding was, that we use `optional` for `function_index`, similar to `optional int32 mapping_index = 1;` to have consistency. And improve on the documentation of the fields within `message Function`, pointing to the special meaning of `Profile.string_table[0]`.

Reply:
I don't see it written down in the agenda but I do remember the point of "specialness" for tracking a missing value for strings was raised and the final consensus was to use `optional`. Maybe @felixge can chime in.

Reply:
Unfortunately I don't recall if we decided to change how we handle strindex fields (we should capture notes better), but thinking more about this I would propose we leave them without optional keyword and with special zero index value meaning empty for now. We can discuss later.

Motivation: optional field at least in Go are encoded at runtime as a field by pointer rather than by value, so if we make them all optional we end up with a lot more "*int32" fields instead of "int32". These dynamic allocations can add up and have memory usage and GC CPU costs.

Reply:
Based on the feedback around the `optional` field in https://cloud-native.slack.com/archives/C01N7PP1THC/p1741664907074709 (There is a historical reason why OTel sparsely make use of `optional`) I have applied the suggestions from https://github.com/open-telemetry/opentelemetry-proto/pull/621#discussion_r1947132154 and set `optional` only for int64 values.